sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/text/TextSuite.scala

```diff
@@ -57,6 +57,13 @@ class TextSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("compression") {
+    val tempDirPath = Utils.createTempDir().getAbsolutePath
+    val df = sqlContext.read.text(testFile)
+    df.write.option("compressionCodec", "GzipCodec").mode(SaveMode.Overwrite).text(tempDirPath)
```

Comment:
what are the mappings provided here? I personally find "GzipCodec" pretty annoying to use. Maybe we can create short keys for these, similar to how we handle compression elsewhere.